inspirehep/modules/orcid/utils.py

```diff
@@ -143,3 +148,32 @@ def get_push_access_token(orcid):
     token, _ = remote_token.token()
 
     return token
+
+
+def get_orcids_for_push(record):
+    """Obtain the ORCIDs associated to the list of authors in the Literature record.
+
+    The ORCIDs are looked up both in the ``ids`` of the ``authors`` and in the
+    Author records that have claimed the paper.
+
+    Args:
+        record(dict): metadata from a Literature record
+
+    Returns:
+        Iterator[str]: all ORCIDs associated to these authors
+    """
+    orcids_on_record = []
+    author_recids_with_claims = []
+
+    for author in record.get('authors', []):
+        orcids_in_author = get_values_for_schema(author.get('ids', []), 'ORCID')
+        if orcids_in_author:
```

Comment:
Minor: this if is unnecessary

Reply:
it's not, because of the `elif` later. I could transform the `elif [clause]` into `if not orcids_in_author and [clause]`, but I think that's less readable.